src/DatadirTestCase.php

```diff
@@ -0,0 +1,194 @@
+     */
+    public function testDatadir(DatadirTestSpecificationInterface $specification): void
+    {
+        $fs = new Filesystem();
+
+        $script = $this->getScript();
+        if (!$fs->exists($script)) {
+            throw new Exception(sprintf(
+                'Cannot open script file "%s"',
+                $script
+            ));
+        }
+
+        $temp = new Temp();
+        $temp->initRunFolder();
+
+        $fs->mirror($specification->getWorkingDirectory() . '/source/data', $temp->getTmpFolder());
+
+        $fs->mkdir($temp->getTmpFolder() . '/in/tables', 0777);
+        $fs->mkdir($temp->getTmpFolder() . '/in/files', 0777);
+        $fs->mkdir($temp->getTmpFolder() . '/out/tables', 0777);
+        $fs->mkdir($temp->getTmpFolder() . '/out/files', 0777);
+
+        $runCommand = [
+            "php",
+            $script,
+        ];
+        $runProcess = new Process($runCommand);
+        $runProcess->setEnv([
+            'KBC_DATADIR' => $temp->getTmpFolder(),
+        ]);
+        $runProcess->run();
+
+        if ($specification->getExpectedReturnCode() !== null) {
+            $this->assertProcessReturnCode($specification->getExpectedReturnCode(), $runProcess);
+        } else {
+            $this->assertNotSame(0, $runProcess->getExitCode(), 'Exit code should have been non-zero');
+        }
+        if ($specification->getExpectedStdout() !== null) {
+            $this->assertSame(
+                $specification->getExpectedStdout(),
+                $runProcess->getOutput(),
+                'Failed asserting stdout output'
+            );
+        }
+        if ($specification->getExpectedStderr() !== null) {
+            $this->assertSame(
+                $specification->getExpectedStderr(),
+                $runProcess->getErrorOutput(),
+                'Failed asserting stderr output'
+            );
+        }
+        if ($specification->getExpectedOutputDirectory() !== null) {
+            $this->assertDirectoryContentsSame(
+                $specification->getWorkingDirectory() . '/expected/data/out',
+                $temp->getTmpFolder() . '/out'
+            );
+        }
+    }
+
+    /**
+     * @return DatadirTestSpecificationInterface[][]
+     */
+    public function provideDatadirSpecifications(): array
+    {
+        $data = [[]];
+        foreach ($this->getDataProviders() as $provider) {
+            $data[] = $provider();
+        }
+        // https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop
+        return array_merge(...$data);
+    }
+
+    /**
+     * @return DatadirTestsProviderInterface[]
+     */
+    protected function getDataProviders(): array
+    {
+        return [
+            new DatadirTestsFromDirectoryProvider($this->getTestFileDir()),
+        ];
+    }
+
+    protected function getScript(): string
+    {
+        return $this->getTestFileDir() . '/../../src/run.php'; // TODO: Change the autogenerated stub
```

Comment:
co je myšleno tímhle TODO?

Reply:
Že jsem lempl a nesmazal jsem to :)

Reply:
Generuje PhpStorm, když zavoláš override...

Reply:
aha díky, override z PhpStorm jsem ještě nepoužíval... už to vidím.